friends/src/App.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import { Route, withRouter } from 'react-router-dom';
+
+import './App.css';
+
+import Nav from './components/Nav';
+import Friends from './components/Friends';
+import Friend from './components/Friend';
+import FriendForm from './components/FriendForm';
+
+import styled from 'styled-components';
+
+const Img = styled.img`
+    padding-top: 20px;
+    opacity: .5;
+    filter: alpha(opacity=50);
+
+    &:hover {
+      opacity: 1;
+      filter: alpha(opacity=100);
+      transition: opacity .5s;
+    }
+`;
+
+class App extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+        friends: [],
+    }
+}
+
+  componentDidMount() {
+    axios
+        .get('http://localhost:5000/friends')
+        .then(response => {
+            this.setState({ friends : response.data })
+        })
+        .catch(err => {
+            console.log(err)
+        })
+  };
+
+  addFriend = (newFriend) => {
+    axios
+      .post('http://localhost:5000/friends', newFriend)
+      .then(response => 
+        this.setState({ friends : response.data }));
+        this.props.history.push('/friends');
+  }
+
+  // updateFriend = (id) => {
+  //   axios
+  //     .post(`http://localhost:5000/friends/${id}`)
+  //     .then(response =>
+  //       )
+  // }
+
+  deleteFriend = (id) => {
+    axios
+      .delete(`http://localhost:5000/friends/${id}`)
+      .then(response =>
+        this.setState({ friends : response.data }));
+        this.props.history.push('/friends');
+  }
+
+
+  render() {
+    console.log(this.state.friends)
+    return (
+      <div className="App">
+        <Nav />
+        <Route exact path="/friends" render={ props => (
+          <Friends {...props} friends={this.state.friends} />
+        )} />
+        <Route path="/friends/:id" render={ props => (
+          <Friend {...props} friends={this.state.friends} delete={this.deleteFriend} />
```

Comment:
consider more differentiation between the names of your components to minimize possible confusion 😕 

Beware the `s`! ⚠️